persistence/play-cqrs/scaladsl/src/main/scala/play/scaladsl/cqrs/EntityFactory.scala

```diff
@@ -19,15 +19,60 @@ import scala.reflect.ClassTag
 import akka.annotation.ApiMayChange
 import akka.cluster.sharding.typed.ShardingEnvelope
 
+trait EntityDef[Command, Event, State] {
+  protected type CommandHandler = (State, Command) => ReplyEffect[Event, State]
+  protected type EventHandler = (State, Event) => State
+
+  def emptyState: State
+  def commandHandler: CommandHandler
+  def eventHandler: EventHandler
+  def tagger: Tagger[Event]
+}
```

Comment:
what is this? It should be possible to define an EventSourcedBehavior in the same way in Akka and Play, to be able to move between those. This defeats that purpose. If we want some glue code for dependency injection and sharding initialization that can be fine, but not another way of defining the behavior.

Reply:
This is only glueing the Tagger to the model and moving to an inheritance style over functional style. I think we can pass it. We can also create confusion with `EntityRef` and `EntityDef`. 

The Java API for EventSourcedBehavior is already using OO. To do the same will have to extend two classes. The behavior itself and then EntityDef without much added value.